### PR TITLE
Update dints template to improve gpu utilization rate

### PR DESCRIPTION
**Pull request by dongyang0122**
Update dints template to improve gpu utilization rate

### CODE REVIEW

File: auto3dseg/algorithm_templates/dints/scripts/train.py [hunk 1]
```diff
@@ -148,30 +157,30 @@ def run(config_file: Optional[Union[str, Sequence[str]]] = None, **override):
...
+        progress=True,
     )
     val_ds = monai.data.CacheDataset(
         data=val_files,
         transform=val_transforms,
         cache_rate=validate_cache_rate,
         hash_as_key=True,
         num_workers=parser.get_parsed_content("training#num_cache_workers"),
-        progress=False,
+        progress=True,
```
Comment (mingxin-zheng) [inline auto3dseg/algorithm_templates/dints/scripts/train.py:168]: Same comment as above.

File: auto3dseg/algorithm_templates/dints/scripts/train.py [hunk 2]
```diff
@@ -148,30 +157,30 @@ def run(config_file: Optional[Union[str, Sequence[str]]] = None, **override):
...
+        persistent_workers=True,
+        pin_memory=True,
     )
-    val_loader = DataLoader(val_ds, num_workers=0, batch_size=1, shuffle=False)
-
-    device = (
-        torch.device(f"cuda:{os.environ['LOCAL_RANK']}")
-        if world_size > 1
-        else torch.device("cuda:0")
+    val_loader = DataLoader(
```
Comment (mingxin-zheng) [inline auto3dseg/algorithm_templates/dints/scripts/train.py:179]: Maybe we should make `num_workers`, `batch_size`, `shuffle`, `persistent_workers`, `pin_memory` all configurable by configs to adapt different environments.

File: auto3dseg/algorithm_templates/dints/scripts/train.py [hunk 3]
```diff
@@ -324,29 +305,22 @@ def run(config_file: Optional[Union[str, Sequence[str]]] = None, **override):
                 f"best mean dice: {best_metric:.4f} at epoch {best_metric_epoch}"
             )
 
+        del inputs, labels, outputs
```
Comment (mingxin-zheng) [inline auto3dseg/algorithm_templates/dints/scripts/train.py:308]: Does this manual deletion that PyTorch is unable to release the GPU mem if they are allocated in another process?

File: auto3dseg/algorithm_templates/dints/scripts/train.py [hunk 4]
```diff
@@ -148,30 +157,30 @@ def run(config_file: Optional[Union[str, Sequence[str]]] = None, **override):
         cache_rate=train_cache_rate,
         hash_as_key=True,
         num_workers=parser.get_parsed_content("training#num_cache_workers"),
-        progress=False,
+        progress=True,
```
[the remark this answers is not in the document]
Comment (dongyang0122) [reply auto3dseg/algorithm_templates/dints/scripts/train.py:160]: Agree, we should make it optional. It is necessary to be true for debugging purposes.

File: auto3dseg/algorithm_templates/dints/scripts/train.py (hunk 1 above)
Comment (dongyang0122) [reply auto3dseg/algorithm_templates/dints/scripts/train.py:168]: Same comment as above. :)

File: auto3dseg/algorithm_templates/dints/scripts/train.py (hunk 3 above)
Comment (dongyang0122) [reply auto3dseg/algorithm_templates/dints/scripts/train.py]: what do you mean here?

Comment (mingxin-zheng) [reply auto3dseg/algorithm_templates/dints/scripts/train.py]: @dongyang0122 I forgot what I was looking for. Please ignore it.

File: auto3dseg/algorithm_templates/dints/scripts/train.py (hunk 2 above)
Comment (dongyang0122) [reply auto3dseg/algorithm_templates/dints/scripts/train.py:286]: I added `num_workers` for validation data loader. For other options, they either cannot be changed (like batch size), or do not make any significant difference when the user changes them.

**Review by mingxin-zheng — APPROVED** [review]
